pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -144,6 +147,9 @@ func NewAlertmanager(ctx context.Context, orgID int64, cfg *setting.Cfg, store A
 		decryptFn:           decryptFn,
 		fileStore:           fileStore,
 		logger:              l,
+
+		// TODO: Preferably, logic around autogen would be outside of the specific alertmanager implementation so that remote alertmanager will get it for free.
+		withAutogen: withAutogen,
```

Comment:
Let's create an issue to not forget about it, and reference PR that refactors Alertmanager

Reply:
https://github.com/grafana/grafana/issues/82042